beacon-chain/core/light-client/store.go

```diff
@@ -146,7 +156,32 @@ func (s *Store) SaveLightClientUpdate(ctx context.Context, period uint64, update
 func (s *Store) SetLastFinalityUpdate(update interfaces.LightClientFinalityUpdate) {
```

Comment:
This functions needs more comprehensive tests. I commented the following three pieces of code and all tests still pass:
```
s.lastFinalityUpdate.FinalizedHeader().Beacon().Slot < update.FinalizedHeader().Beacon().Slot
```
```
if UpdateHasSupermajority(update.SyncAggregate()) {
	s.finalityUpdateNeedsBroadcast = false
} else {
	s.finalityUpdateNeedsBroadcast = true
}
```
```
if s.finalityUpdateNeedsBroadcast && UpdateHasSupermajority(update.SyncAggregate()) {
	if err := s.p2p.BroadcastLightClientFinalityUpdate(context.Background(), update); err != nil {
		log.WithError(err).Error("Could not broadcast light client finality update")
	}
	s.finalityUpdateNeedsBroadcast = false
}
```